### DISCUSSION TITLE
Passing variables from one class to another

### DISCUSSION THREAD

**Discussion by Smithy419** [TomSchimansky/CustomTkinter#1842] [Q&A]
Hi, I’m pretty new to customkinter and I’m wondering how do I pass a string variable from one class to another when both classes are customtkinter.CTkTopLevel?

 It’s more confusing than normal tkinter for me because of the *args ** kwargs stuff 

thanks

**Comment by avalon60** [comment]
Hi @Smithy419.

Are you wanting this to work in one direction? If so when you instantiate the second CTkTopLevel window, you can pass the first CTkTopLevel subclass object as a parameter. Then from the second CTkTopLevel, you would then have access to any methods / properties defined for the CTkTopLevel. It's then just a case of having a method which receives the string, and does with it as you wish. You can even update a widget directly, with the string, as long as the widget is defined with self.

Alternatively (but not as nice), if you pass a common master widget (typically the CTk widget):

```
top_level1 = TopLevel1(master=root_window)
...
top_level2 = TopLevel2(master=root_window)
...
```

You can cross reference one of the TopLevel_n_, by using something like:

```
class TopLevel2(CTkTopLevel):
...
self.master.top_level1...
```
This assumes that both CTkTopLevel subclasses are siblings.

**Reply by Smithy419** [reply]
Hi, thanks for your reply. I am creating a booking system and there's a lot of info I need to put into an SQL database. Basically I want the main CTk Toplevel window where the booking actually takes place is gonna take the values from a secondary window which opens  when a button is clicked on the main window. Not sure if this is the most efficient way to do it but oh well. 

"Are you wanting this to work in one direction?" - Yes, but I am not sure which CTk Toplevel to instantiate  the variable to. Do I only instantiate the variable to the Toplevel window I want to pass it to? I've been trying the first method of your reply and I am getting an error saying the variable is not defined when it clearly has been defined in the other toplevel window.

I'm relatively new to OOP and customtkinter and the solution probably is simple to get to, but could you further help me?

Thanks

**Reply by avalon60** [reply]
I'm not clear whether your CTkTopLevels are siblings or whether one is launching the other. I think you are saying that your root window (CTk()) is launching both, so let's take it from there...

So in your root window, you do something like:


```
root = CTk()
...
booking_window = MyBookingWindow(master=root)
```
Let's assume your MyBookingWindow class has a method, called process_variable:

```
def process_variable(myvar):
    self.myvar = myvar
    # do something else, for example:
    label_widget.configure(text=self.myvar)
```
Subsequently,  button pressed and we launch second window...

`second_window = MySecondWinClass(master=root, booking_win=booking_window)`
So note that in the above, we are passing a reference to the booking window to the secondary window instance.

Now from within your MySecondWinClass code, you can call any methods defined in the MyBookingWindow class:

`booking_win.process_variable(myvar=value)`

This assumes that your MyBookingWindow class includes something like:

```
class MyBookingWindow(CTkTopLevel):

    def __init__(self, booking_win, *args, **kwargs):
        super().__init__(*args, **kwargs)

```

NOTE: If instead, you are defining your root window as a subclass of CTk(), then instead of passing "master=root", you would pass "master=self".
